gateway-spi/src/main/java/org/apache/knox/gateway/services/security/SSLService.java

```diff
@@ -22,8 +22,9 @@
 import java.security.NoSuchAlgorithmException;
 import java.security.cert.CertificateException;
 
+import org.apache.knox.gateway.config.GatewayConfig;
 import org.apache.knox.gateway.services.Service;
 
 public interface SSLService extends Service {
-  Object buildSslContextFactory(String keystoreFileName, String keystoreType, String alias) throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException;
+  Object buildSslContextFactory(GatewayConfig config) throws KeyStoreException, IOException, CertificateException, NoSuchAlgorithmException, AliasServiceException;
```

Comment:
Question - Is there an exception that `KeystoreException`, `CertificateException`, `NoSuchAlgorithmException` all extend from? Seems weird to have to call them out individually. I don't think we should replace with just plain `Exception` either.

Reply:
Looks like that should have been reduced to just `throws AliasServiceException`.  I forgot to check and assumed my IDE remove it from the interface method when I _smart_ removed it from the implementation.

I will fix this.